ext/src/main/java/io/burt/kafka/clients/KafkaClientsLibrary.java

```diff
@@ -58,6 +59,8 @@ static RubyClass mapErrorClass(Ruby runtime, Throwable t) {
       String javaName = t.getClass().getSimpleName();
       String rubyName = javaName.substring(0, javaName.length() - 9) + "Error";
       return (RubyClass) runtime.getClassFromPath(String.format("Kafka::Clients::%s", rubyName));
+    } else if (t instanceof TimeoutException) {
+      return (RubyClass) runtime.getModule("Timeout").getConstant("Error");
```

Comment:
Yes add a new error for this. Any use of the timeout library should be avoided. It's among the worst parts of the standard library. I don't know why I even suggested loading it because the idea makes me shudder.